driver-sync/src/test/functional/com/mongodb/client/CommandMonitoringTest.java

```diff
@@ -175,118 +142,11 @@ private ServerVersion getServerVersion(final String fieldName) {
-            response.put("n", new BsonInt32(response.getNumber("n").intValue()));
-        }
-
-        if (actual.getCommandName().equals("find") || actual.getCommandName().equals("getMore")) {
-            if (response.containsKey("cursor")) {
-                if (response.getDocument("cursor").containsKey("id")
-                    && !response.getDocument("cursor").getInt64("id").equals(new BsonInt64(0))) {
-                    response.getDocument("cursor").put("id", new BsonInt64(42));
-                }
-            }
-        } else if (actual.getCommandName().equals("killCursors")) {
-            response.getArray("cursorsUnknown").set(0, new BsonInt64(42));
-        } else if (isWriteCommand(actual.getCommandName())) {
-            if (response.containsKey("writeErrors")) {
-                for (Iterator<BsonValue> iter = response.getArray("writeErrors").iterator(); iter.hasNext();) {
-                    BsonDocument cur = iter.next().asDocument();
-                    cur.put("code", new BsonInt32(42));
-                    cur.put("errmsg", new BsonString(""));
-                    cur.remove("codeName");
-                }
-            }
-            if (actual.getCommandName().equals("update")) {
-                response.remove("nModified");
-            }
-        }
-        return new CommandSucceededEvent(actual.getRequestId(), actual.getConnectionDescription(), actual.getCommandName(), response,
-                actual.getElapsedTime(TimeUnit.NANOSECONDS));
-    }
-
-    private boolean isWriteCommand(final String commandName) {
-        return asList("insert", "update", "delete").contains(commandName);
-    }
-
-    private CommandStartedEvent massageActualCommandStartedEvent(final CommandStartedEvent actual) {
-        BsonDocument command = getWritableCloneOfCommand(actual.getCommand());
-
-        if (actual.getCommandName().equals("update")) {
-            for (Iterator<BsonValue> iter = command.getArray("updates").iterator(); iter.hasNext();) {
-                BsonDocument curUpdate = iter.next().asDocument();
-                if (!curUpdate.containsKey("multi")) {
-                    curUpdate.put("multi", BsonBoolean.FALSE);
-                }
-                if (!curUpdate.containsKey("upsert")) {
-                    curUpdate.put("upsert", BsonBoolean.FALSE);
-                }
-            }
-        } else if (actual.getCommandName().equals("getMore")) {
-            command.put("getMore", new BsonInt64(42));
-        } else if (actual.getCommandName().equals("killCursors")) {
-            command.getArray("cursors").set(0, new BsonInt64(42));
-        }
-        command.remove("$clusterTime");
-        command.remove("lsid");
-
-        return new CommandStartedEvent(actual.getRequestId(), actual.getConnectionDescription(), actual.getDatabaseName(),
-                actual.getCommandName(), command);
+        CommandMonitoringTestHelper.assertEventsEquality(expectedEvents, events);
```

Comment:
Refactored this so that it can be shard with the transaction tests, which also assert on command events